cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java

```diff
@@ -423,6 +481,37 @@ private void performUpgrade() throws Exception {
       metricsKafkaUpgrader.upgrade();
       hBaseTableUtil.dropTable(hBaseAdmin, metricsKafkaUpgrader.getOldKafkaMetricsTableId());
     }
+
+    upgradeAdapters();
+  }
+
+  /**
+   * Upgrades adapters by first deleting all the schedule triggers and the the job itself in every namespace follwed by
+   * all the adapters in the namespace
+   *
+   * @throws SchedulerException
+   */
+  private void upgradeAdapters() throws SchedulerException {
+    DatasetBasedTimeScheduleStore datasetBasedTimeScheduleStore = getDatasetBasedTimeScheduleStore();
+    NamespaceAdmin namespaceAdmin = injector.getInstance(NamespaceAdmin.class);
+    List<NamespaceMeta> namespaceMetas = namespaceAdmin.listNamespaces();
+    for (NamespaceMeta namespaceMeta : namespaceMetas) {
+      Collection<AdapterSpecification> adapters = getAdapterService().getAdapters(Id.Namespace
+                                                                                    .from(namespaceMeta.getName()));
```

Comment:
nit: namespaceMeta.getName() is used several places, would be less verbose to store it in a variable.